### ISSUE TITLE
Namespace package not detected if installed in site-packages

### ISSUE THREAD

**Issue by DingXuefeng** [pylint-dev/astroid#2635]
pylint does not respect namespace pacakge. after debugging I found it's because `is_namespace` function defined in `astroid.interpreter._import.util.py` returns `False` if the `submodule search location` is in `STD_AND_EXT_LIB_DIRS`.
```python
        if found_spec and found_spec.submodule_search_locations:
            # But immediately return False if we can detect we are in stdlib
            # or external lib (e.g site-packages)
            if any(
                any(location.startswith(lib_dir) for lib_dir in STD_AND_EXT_LIB_DIRS)
                for location in found_spec.submodule_search_locations
            ):
                return False
```
this is not a valid assumption. PEP 420 does not say this.

We are working on a big project as many teams. Our project lives in a namespace, say `myproj`. Different groups work on subproject, such as `myproj.simulation` and `myproj.plot` and `myproj.calculation`. These 3 packages are 3 different git repo and can be `pip` installed separately but they all goes to `site-packages/myproj` folder as `site-packages/myproj/simulation`, `site-pacakges/myproj/plot` etc. the `myproj` itself is a namespace package.

We have this need and I believe this is compatible with PEP420, but not with astroid.

I suggest remove line 96-102 of `astroid/interpreter/_import/util.py`.

**Comment by jacobtylerwalls** [comment]
We can't just do a straight revert of #1756 without having another solution in hand for not recausing pylint-dev/pylint#7365.

**Comment by DingXuefeng** [comment]
> We can't just do a straight revert of [#1756](https://github.com/pylint-dev/astroid/pull/1756) without having another solution in hand for not recausing [pylint-dev/pylint#7365](https://github.com/pylint-dev/pylint/issues/7365).

Thanks, and I get it. It seems it's not easy to fix it. Let me see if I can fix both with a PR.

**Comment by DingXuefeng** [comment]
Hi, unfortunately we cannot afford such debug. We decide to move the design of the project into a single package to make life easier. Since the need vanish, we can close this issue. Thanks for your reading.
